### ISSUE TITLE
Secondary monitor recognized but does not display workspace on multi-GPU intel+nvidia setup

### ISSUE THREAD

**Issue by tannart** [swaywm/sway#6246]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions on IRC at #sway on irc.freenode.net.
- Proprietary graphics drivers, including nvidia, are not supported. Please use the open source equivalents, such as nouveau, if you would like to use Sway.
- Please do NOT submit issues for information from the github wiki. The github wiki is community maintained and therefore may contain outdated information, scripts that don't work or obsolete workarounds.
  If you fix a script or find outdated information, don't hesitate to adjust the wiki page.

### Please fill out the following:
- **Sway Version:**
  - `swaymsg -t get_version` or `sway -v`

sway version 1.6-7c74f01f (May  1 2021, branch 'master')

- **Debug Log:**
  - Run `sway -d 2> ~/sway.log` from a TTY and upload it to a pastebin, such as gist.github.com.
  - This will record information about sway's activity. Please try to keep the reproduction as brief as possible and exit sway.

https://paste.sr.ht/~tannart/f59fe382594139a445d7f6f49f1a4448db99c22a

- **Configuration File:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.

Used default configuration
https://paste.sr.ht/~tannart/b7ac75be4e2058c18955b71219b142064c7fb2d2

- **Stack Trace:**
  - This is only needed if sway crashes.
  - If you use systemd, you should be able to open the coredump of the most recent crash with gdb with
    `coredumpctl gdb sway` and then `bt full` to obtain the stack trace.
  - If the lines mentioning sway or wlroots have `??` for the location, your binaries were built without debug symbols. Please compile both sway and wlroots from source and try to reproduce.
No crash

- **Description:**
  - The steps you took in plain English to reproduce the problem.

On starting sway the secondary monitor shows a black display, however workspaces are assigned to it and it does display the cursor if you mouse over it and occasionally waybar will be shown flickering at the top of the screen.

The output used is HDMI-A-4, unplugging and plugging the monitor does not resolve the issue. The monitor works fine using Gnome Wayland.

**Comment by Ethan-ks** [comment]
I am also seeing this same behavior, as of a dew days ago. In my case, I'm using a thunderbolt -> laptop dock -> hdmi monitor. I checked out the v1.5 branch (and an appropriate version (0.12.0 via commit 238d1c0) of wlroots) and this fixed the issue, I can once again see the workspaces on my second display.

Let me know if there is anything I can do to provide more information/help find the bug. I'm not super good with git, but I can follow instructions you provide.

**Comment by emersion** [comment]
```
00:00:00.059 [ERROR] [wlr] [EGL] command: eglGetPlatformDisplay, error: EGL_BAD_PARAMETER (0x300c), message: "EGL_BAD_PARAMETER error: In internal function: Additional INFO may be available
"
00:00:00.059 [INFO] [wlr] [EGL] command: eglGetPlatformDisplay, error: EGL_BAD_PARAMETER (0x300c), message: "EGL_BAD_PARAMETER error: In function eglGetPlatformDisplay(), unsupported EGL Platform
"
00:00:00.064 [ERROR] [wlr] [EGL] command: eglQueryDevicesEXT, error: EGL_BAD_ALLOC (0x3003), message: "EGL_BAD_ALLOC error: In eglQueryDevicesEXT: Failed to allocate device list.
"
```

To fix this one, upgrade to the latest glvnd release or uninstall the proprietary nvidia drivers.

> On starting sway the secondary monitor shows a black display, however workspaces are assigned to it and it does display the cursor if you mouse over it and occasionally waybar will be shown flickering at the top of the screen.

To fix this, have you tried `WLR_DRM_NO_MODIFIERS=1` as described in the wiki?

**Comment by nocoolnametom** [comment]
Running sway 1.6 with `WLR_DRM_NO_MODIFIERS=1` and up-to-date drivers (currently checking if the kernel param `nvidia-drm.modeset=1` is also needed or not) fixed the problem for me, though it's still a bit odd if sway starts with the second monitor already attached.  Plugging it in after sway has started, though, and it's back to working as normally!  I use NixOS and had been starting every day loading an older configuration from April 6 and wondering what broke between then and April 9. Thanks a ton!

**Comment by tannart** [comment]
> To fix this, have you tried WLR_DRM_NO_MODIFIERS=1 as described in the wiki?

This resolved the issue for me, thanks so much! Not sure how
